### DISCUSSION TITLE
Usage of SSIM loss in binary segmentation

### DISCUSSION THREAD

**Discussion by rose-jinyang** [kornia/kornia#1600] [Questions] [answered]
Hello
how are you?
Thanks for contributing to this project.
Can we use the SSIM loss in the binary image segmentation?
Thanks

**Comment by edgarriba** [comment, marked as answer]
hi, @rose-jinyang  -- yes, if you check the [ssim_loss](https://kornia.readthedocs.io/en/latest/losses.html?highlight=ssim%20loss#kornia.losses.ssim_lossl) will see that the operator is agnostic to the number of channels.

```python
import kornia
import torch

device = torch.device("cpu")
dtype = torch.float64

B, C, H, W = 1, 1, 32, 32

image1 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
image2 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)

loss = kornia.losses.ssim_loss(image1, image2, window_size=5)
print(loss)
```

**Comment by rose-jinyang** [comment]
Thanks
